### PR TITLE
Update developer.md

### PR DESCRIPTION
**Pull request by janosSarusiKis**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets | 
| License         | Apache 2.0

### What's in this PR?
Small changes for developer documentation.

### Why?
The dex.yml was created in a different location, and there is no mentioned sections in config.yml.

### Checklist
- [ x] User guide and development docs updated

### CODE REVIEW

File: docs/developer.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ which removes everything.
 
 ### Configuration
 
-Create a `config/config.yaml` and `config/dex.yml` config file based on their `config/*.dist` counterparts with:
+Create a `config/config.yaml` and `etc/config/dex.yml` config file based on their `config/*.dist` counterparts with:
 
 ```bash
 $ make config/config.yaml config/dex.yml
```
Comment (sagikazarmark) [inline docs/developer.md:44]: I assume the make commands are also outdated. Can you please check them?

Comment (janosSarusiKis) [reply docs/developer.md:44]: They are ok, it was wrong only in the documentation.

Comment (sagikazarmark) [reply docs/developer.md:44]:
```suggestion
$ make config/config.yaml etc/config/dex.yml
```

**Comment by janosSarusiKis** [comment]
> - Use static Email/Password authentication following the example in `config/dex.yml.example` (staticPasswords sections)

Should this be completely removed or changed to etc/config/dex.yml ?

**Comment by sagikazarmark** [comment]
@janosSarusiKis I guess it should be changed.

**Pull request merged**